src/main/crc.ts

```diff
@@ -0,0 +1,54 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import { crc32 } from "crc";
+import * as yauzl from "yauzl";
+
+type TCrcFile = [string, string];
+
+export async function extractCrc32OnZip(filePath: string) {
+    const fileArray: TCrcFile[] = [];
+    return new Promise<string>((resolve, reject) => {
+
+        yauzl.open(filePath, { lazyEntries: true }, (err, data) => {
+            if (err) {
+                reject(err);
+            }
+            data.on("error", (e) => reject(e));
+            data.readEntry();
+            data.on("entry", (entry: yauzl.Entry) => {
+                fileArray.push([entry.fileName, (entry.crc32 || 0).toString(16)]);
```

Comment:
Based on my test, with some EPUBs `yauzl` returns `0` CRC for some ZIP directory entries.
Therefore I would suggest filtering out zero values (e.g. `arr.filter((val) => { return val; })`), in order to be consistent with other ZIP libs such as `unzipper` and `node-stream-zip`.

See https://github.com/readium/readium-desktop/pull/793#issuecomment-542363462

Reply:
Good point!